Java: add documentation for the various packages (#3342)
microsoft/semantic-kernel: java/PACKAGES.md

```xml
    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>com.microsoft.semantic-kernel</groupId>
                <artifactId>semantickernel-bom</artifactId>
                <version>${semantickernel.version}</version>
                <scope>import</scope>
                <type>pom</type>
            </dependency>
        </dependencies>
    </dependencyManagement>
```

## Common Packages

`semantickernel-bom`
: A BOM that can be used to define the versions of all semantic kernel packages.

`semantickernel-api`
: Main API for the semantic kernel. Defines the core public API for the semantic kernel.

`semantickernel-core`
: An implementation of the semantic kernel API. This package contains the core implementation of the semantic kernel. This
should be made available to the application at runtime, however these classes are considered internal and should not be used directly
by the application as they may change without notice.

## Connectors

`semantickernel-connectors-ai-openai`
: Provides a connector that can be used to interact with an OpenAI API.

### Memory Connectors

#### JDBC Memory Connectors

> Provides a memory connector that can be used to interact with a JDBC database.

Reviewer: Do we want to make a brief mention of the JDBC package, and how one might want to  use it to extend a JDBC-based connection that we don't provide out of the box?

Reviewer 2: Yes, eventually we will need detailed documentation for each concept, similarly to how Spring Project is documented.

Example of Spring Data JDBC https://docs.spring.io/spring-data/jdbc/docs/current/reference/html/

- `semantickernel-connectors-memory-sqlite`
- `semantickernel-connectors-memory-postgresql`
- `semantickernel-connectors-memory-mysql`

`semantickernel-connectors-memory-azurecognitivesearch`

> : Provides a memory connector that allows using Azure Cognitive Search as a memory.

Reviewer: ```suggestion
: Provides a memory connector for using Azure Cognitive Search as the memory provider for an application.
```

### Miscellaneous Packages

`semantickernel-planners`

> : Implementations of various planners that can be used to plan the execution of semantic functions.

Reviewer: ```suggestion
: Implementations of various planners that can be used in the execution of semantic functions.
```

`semantickernel-plugin-core`

> : Several example plugins, many of which are used in the samples.

Reviewer: ```suggestion
: Several example plugins, many of which are used in the [Java Semantic Kernel samples](/java-development/java/samples).
```

Reviewer 2: "Semantic Kernel for Java" is what we use at all times.
